src/scrapers/esmeraldatorrentScraper.ts

```diff
@@ -0,0 +1,245 @@
+import { load } from 'cheerio';
+import { slugify } from '../lib/strings';
+import Scraper from './scraper';
+import { Magnet, ScraperRequest } from '../interfaces';
+
+const SOURCE = 'EsmeraldaTorrent';
+const DEFAULT_URL = 'https://esmeraldatorrent.com';
+
+export class EsmeraldatorrentScraper extends Scraper {
+
+  baseUrl: string = process.env.BASE_URL || DEFAULT_URL;
+
+  headers: Record<string, string>;
+
+  constructor() {
+    super('esmeraldatorrent');
+    this.headers = {
+      authority: this.baseUrl.split('/')[2],
+      'sec-ch-ua':
+        '" Not A;Brand";v="99", "Chromium";v="90", "Google Chrome";v="90"',
+      'sec-ch-ua-mobile': '?0',
+      'upgrade-insecure-requests': '1',
+      'user-agent':
+        'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.93 Safari/537.36',
+      accept:
+        'text/html,application/xhtml+xml,application/xml;' +
+        'q=0.9,image/avif,image/webp,image/apng,*/*;' +
+        'q=0.8,application/signed-exchange;v=b3;q=0.9',
+      'sec-fetch-site': 'same-origin',
+      'sec-fetch-mode': 'navigate',
+      'sec-fetch-user': '?1',
+      'sec-fetch-dest': 'document',
+      referer: 'FILLED IN LATER',
+      'accept-language': 'es,en-US;q=0.9,en;q=0.8',
+      cookie: 'telegram=true; mode=light',
+    };
+  }
+
+  protected processMessage(message: ScraperRequest): Promise<Magnet[]> {
+    if (!message.title) throw new Error('Title is required');
+    if (message.seasonNum && message.episodeNum) {
+      return this.getEpisodeLinks(
+        message.spanishTitle || message.title,
+        message.seasonNum,
+        message.episodeNum,
+      );
+    } else {
+      return this.getMovieLinks(
+        message.spanishTitle || message.title,
+        message.year,
+      );
+    }
+  }
+
+  async getMovieLinks(title: string, year: number): Promise<Magnet[]> {
+    try {
+      const encodedTitle = encodeURI(title);
+      const searchUrl = `${this.baseUrl}/buscar/${encodedTitle}`;
+      const result = await fetch(searchUrl, {
+        method: 'GET',
+        headers: { ...this.headers, referer: searchUrl },
+      });
+      const text = await result.text();
+      const $ = load(text);
+      const items = $('div#buscador div div p span a').toArray();
+      const results: Magnet[] = [];
+      for (const value of items) {
+        const foundTitle = slugify($(value).text());
+        if (foundTitle === slugify(title)) {
+          const magnet = await this.getMovieInfo($(value).attr('href') as string, year);
+          if (magnet) results.push(magnet);
+        }
+      }
+      return results;
+    } catch (error) {
+      console.log('DT - ERROR', error);
```

Comment:
[Sand] DT stands for DonTorrent, we could use ET or EsmeraldaTorrent for this one